canvas_modules/common-canvas/src/common-properties/controls/checkboxset/checkboxset.jsx

```diff
@@ -74,27 +78,54 @@ class CheckboxsetControl extends React.Component {
 	}
 
 	render() {
+
+
+		const hidden = this.props.state === STATES.HIDDEN;
+		const disabledTip = this.props.state === STATES.DISABLED;
+
 		let controlValue = this.props.value;
 		if (typeof controlValue === "undefined" || controlValue === null) {
 			controlValue = [];
 		}
 		const checkboxes = [];
 		for (var i = 0; i < this.props.control.values.length; i++) {
+			let tooltip = "";
+			if (this.props.control.valueDescs && !this.props.tableControl) {
+				tooltip = (
+					<span >{this.props.control.valueDescs[i]}</span>
+				);
+			}
+			const tooltipIcon = isEmpty(tooltip) ? "" : (this.props.control.valueDescs[i].indexOf("desc") !== -1) && (
```

Comment:
Why are you testing this ` (this.props.control.valueDescs[i].indexOf("desc") !== -1)`?  Wouldn't that only be true if the description has `desc` in the string?  

Would `isEmpty(tooltip)` ever be false?  This is a span above.